sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -65,6 +65,18 @@ trait FileFormat {
   }
 
   /**
+   * Returns concrete column vector class names for each column to be used in a columnar batch
+   * if this format supports returning columnar batch.
+   */
+  def vectorTypes(
+      sparkSession: SparkSession,
```

Comment:
I thought `sparkSession` would be needed because we might want to change the vector type based on some configuration. What do you think?
`dataSchema` might not be needed.